rust/azure_iot_operations_mqtt/tests/mock_receive.rs

```diff
@@ -0,0 +1,482 @@
+
+    // Create receivers for a topic filter
+    // NOTE: Do not actually subscribe here, as it's not necessary for the test
+    let receiver = managed_client.create_unfiltered_pub_receiver();
+
+    qos1_single_receiver_test_logic(mock_server, receiver, "test/subscribe/topic").await;
+}
+
+/// Common test logic for multiple filtered/unfiltered single receiver tests at QoS 0.
+/// Tests that:
+/// - all receivers receive all messages with both recv() and recv_manual_ack()
+/// - no PUBACKs are sent because QoS 0
+/// - no mechanism for sending PUBACKs is exposed
+async fn qos0_multiple_receiver_same_filter_test_logic(
+    mock_server: MockServer,
+    mut receiver1: SessionPubReceiver,
+    mut receiver2: SessionPubReceiver,
+    mut receiver3: SessionPubReceiver,
+    topic_name: &str,
+) {
+    ///////////// Using recv() /////////////
+
+    // Send publish from mock server and all receivers receive it via recv()
+    let proto_publish = mqtt_proto::Publish {
+        topic_name: mqtt_proto::topic(topic_name),
+        packet_identifier_dup_qos: mqtt_proto::PacketIdentifierDupQoS::AtMostOnce,
+        retain: false,
+        payload: bytes::Bytes::from("Publish 1"),
+        other_properties: mqtt_proto::PublishOtherProperties::default(),
+    };
+    let expected_publish = proto_publish.clone().into();
+    mock_server.send_publish(proto_publish);
+    assert_eq!(receiver1.recv().await.unwrap(), expected_publish);
+    assert_eq!(receiver2.recv().await.unwrap(), expected_publish);
+    assert_eq!(receiver3.recv().await.unwrap(), expected_publish);
+
+    // No ack expected for QoS 0
+    mock_server.expect_no_packet();
+
+    ///////////// Using recv_manual_ack() /////////////
+
+    // Send publish from mock server and receive it via recv_manual_ack()
+    let proto_publish = mqtt_proto::Publish {
+        topic_name: mqtt_proto::topic(topic_name),
+        packet_identifier_dup_qos: mqtt_proto::PacketIdentifierDupQoS::AtMostOnce,
+        retain: false,
+        payload: bytes::Bytes::from("Publish 2"),
+        other_properties: mqtt_proto::PublishOtherProperties::default(),
+    };
+    let expected_publish = proto_publish.clone().into();
+    mock_server.send_publish(proto_publish);
+    let response1 = receiver1.recv_manual_ack().await.unwrap();
+    assert_eq!(response1.0, expected_publish);
+    assert!(response1.1.is_none()); // No manual ack mechanism for QoS 0
+    let response2 = receiver2.recv_manual_ack().await.unwrap();
+    assert_eq!(response2.0, expected_publish);
+    assert!(response2.1.is_none()); // No manual ack mechanism for QoS 0
+    let response3 = receiver3.recv_manual_ack().await.unwrap();
+    assert_eq!(response3.0, expected_publish);
+    assert!(response3.1.is_none()); // No manual ack mechanism for QoS 0
+
+    // No ack expected for QoS 0
+    mock_server.expect_no_packet();
+}
+
+#[tokio::test]
+async fn qos0_multiple_identical_filtered_receivers() {
```

Comment:
could be interesting to do a variant of this one with different wildcards as well